.github/workflows/umbral-pre.yml

```diff
@@ -106,7 +106,7 @@ jobs:
         rust:
           - stable
         python:
-          - "3.10"
+          - "3.12"
     steps:
       - uses: actions/checkout@v3
```

Comment:
Side note: the versions of some of these dependent actions should be updated:
- checkout: v4
- setup-python: v4
- download-artifact: v3
...

We can either update these as part of this PR or another one (with issue tracking), but we should update them.